src/main/java/jenkins/branch/MultiBranchProject.java

```diff
@@ -563,9 +584,36 @@ public P getItem(String name) {
             if (item != null) {
                 return item;
             }
+            item = super.getItem(NameMangler.apply(decoded));
+            if (item != null) {
+                return item;
+            }
             // fall through for double decoded call paths
         }
-        return super.getItem(name);
+        P item = super.getItem(name);
+        if (item != null) {
+            return item;
+        }
+        return super.getItem(NameMangler.apply(name));
+    }
+
+    /**
+     * Returns the child job with the specified branch name or {@code null} if no such child job exists.
+     *
+     * @param branchName the name of the branch.
+     * @return the child job or {@code null} if no such job exists or if the requesting user does ave permission to
+     * view it.
+     * @since 2.0.0
+     */
+    @CheckForNull
+    public P getItemByBranchName(@NonNull String branchName) {
+        BranchProjectFactory<P, R> factory = getProjectFactory();
+        for (P p: getItems()) {
```

Comment:
If this is performance-sensitive, could try `getItem(branchName)` first optimistically.